Dockerfile

```diff
@@ -1,16 +1,21 @@
 FROM alpine:latest
 
-RUN apk update && apk upgrade
```

Comment:
This line is missing. Why not ``RUN apk update && apk upgrade && apk add --no-cache ...``?

Reply:
- `apk-update` instruction:
Once you're using the `--no-cache`  argument, it allows you to always get up-to-date packages; So, in that sense, the use of `update` instruction has no effects;

Reference: https://github.com/gliderlabs/docker-alpine/blob/master/docs/usage.md#disabling-cache

- `apk-upgrade` instruction:
If you need to upgrade a specific package for certain reasons, the recommended approach would be bumping `alpine` version for the one that provides you this updated package;
> Avoid RUN apt-get upgrade and dist-upgrade, as many of the “essential” packages from the parent images cannot upgrade inside an unprivileged container. If a package contained in the parent image is out-of-date, contact its maintainers. If you know there is a particular package, foo, that needs to be updated, use apt-get install -y foo to update automatically.

Reference: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run

Reply:
Do you have a chance to take a look at it? Does it still valid to be open?

Reply:
Sorry for letting this hang here....

It was just a matter that I wanted to MAKE SURE that packages are up-to-date. It's just a security baseline, removing one instance in the chain we maybe are dependent on. I gave up on complaining about people doing their work properly, see e.g. a pending thing here https://github.com/alpinelinux/docker-alpine/issues/34 .

I just modified the line to ``RUN apk update && apk upgrade && apk add --no-cache curl bash procps drill git coreutils libidn curl``

```
(1/4) Upgrading musl (1.1.20-r4 -> 1.1.20-r5)
(2/4) Upgrading libcrypto1.1 (1.1.1b-r1 -> 1.1.1d-r0)
(3/4) Upgrading libssl1.1 (1.1.1b-r1 -> 1.1.1d-r0)
(4/4) Upgrading musl-utils (1.1.20-r4 -> 1.1.20-r5)
```

If I omit the statement ``apk upgrade`` those lines are missing.

So I will modify the line, boil down the RUN statements to one and close this.

Thanks for the fruitful discussion.